go-controller/pkg/node/management-port_linux.go

```diff
@@ -280,6 +293,138 @@ func createPlatformManagementPort(interfaceName string, localSubnets []*net.IPNe
+		if err != nil {
+			return fmt.Errorf("failed to remove OVS internal port: %s", stderr)
+		}
+		return nil
+	}
+
+	// It is other representor which was used as management port.
+	// Remove it from the bridge and rename.
+	klog.Infof("Found existing representor management port. Removing it")
+	return unconfigureMgmtRepresentorPort(mgmtPortName)
+}
+
+func unconfigureMgmtRepresentorPort(mgmtPortName string) error {
+	// Get saved port name
+	savedName, stderr, err := util.RunOVSVsctl("--if-exists", "get", "Interface", mgmtPortName, "external-ids:ovn-orig-mgmt-port-rep-name")
+	if err != nil {
+		klog.Warningf("Failed to get external-ds:ovn-orig-mgmt-port-rep-name: %s", stderr)
+	}
+
+	if savedName == "" {
+		// rename to PortName + "-ddmmyyHHMMSS"
+		klog.Warningf("No saved management port representor name, falling back to generic name")
+		savedName = mgmtPortName + time.Now().Format("-010206150405")
+	}
+
+	_, stderr, err = util.RunOVSVsctl("--if-exists", "del-port", "br-int", mgmtPortName)
+	if err != nil {
+		return fmt.Errorf("failed to remove OVS port: %s", stderr)
+	}
+
+	link, err := util.GetNetLinkOps().LinkByName(mgmtPortName)
+	if err != nil {
+		return fmt.Errorf("failed to lookup %s link: %v", mgmtPortName, err)
+	}
+
+	if err := util.GetNetLinkOps().LinkSetDown(link); err != nil {
+		return fmt.Errorf("failed to set link down: %v", err)
+	}
+
+	if err := util.GetNetLinkOps().LinkSetName(link, savedName); err != nil {
+		return fmt.Errorf("failed to rename %s link to %s: %v", mgmtPortName, savedName, err)
+	}
+	return nil
+}
+
+func unconfigurePlatformMgmtInterface(hostSubnets []*net.IPNet, mgmtPortName string) error {
+	link, err := util.GetNetLinkOps().LinkByName(mgmtPortName)
+	if err != nil {
+		if !util.GetNetLinkOps().IsLinkNotFoundError(err) {
+			return fmt.Errorf("failed to lookup %s link: %v", mgmtPortName, err)
+		}
+		return nil
+	}
+
+	klog.Infof("Found existing management interface. Unconfiguring it")
+	ipt4, ipt6, err := getIPTablesForHostSubnets(hostSubnets)
+	if err != nil {
+		return fmt.Errorf("failed to get iptables: %v", err)
+	}
+
+	if err := tearDownInterfaceIPConfig(link, ipt4, ipt6); err != nil {
+		return fmt.Errorf("teardown failed: %v", err)
+	}
+
+	if err := util.GetNetLinkOps().LinkSetDown(link); err != nil {
+		return fmt.Errorf("failed to set %s link down: %v", mgmtPortName, err)
+	}
+
+	savedName := ""
+	if config.OvnKubeNode.Mode != types.NodeModeDPUHost {
+		// Get original interface name saved at OVS database
+		stdout, stderr, err := util.RunOVSVsctl("--if-exists", "get", "Open_vSwitch", ".", "external-ids:ovn-orig-mgmt-port-netdev-name")
+		if err != nil {
+			klog.Warningf("Failed to get external-ds:ovn-orig-mgmt-port-netdev-name: %s", stderr)
+		}
+		savedName = stdout
+	}
+
+	if savedName == "" {
+		// rename to PortName + "-ddmmyyHHMMSS"
+		klog.Warningf("No saved management port netdevice name, falling back to generic name")
```

Comment:
should you print the name of the port in the log message?

Reply:
Yes

Reply:
@DmytroLinkin I think it would be better to also print out the savedName in the logs. Just so we have all the info when debugging.